pkg/system/phase2_creating.go

```diff
@@ -450,6 +450,11 @@ func (r *Reconciler) SetDesiredCoreApp() error {
 	r.CoreApp.Spec.Selector.MatchLabels["noobaa-core"] = r.Request.Name
 	r.CoreApp.Spec.ServiceName = r.ServiceMgmt.Name
 
+	//check if provider mode is enabled and signal the core
+	if util.GetAnnotationValue(r.NooBaa.Annotations, "MulticloudObjectGatewayProviderMode") == "true" {
+		util.GetEnvVariable(&r.CoreApp.Spec.Template.Spec.Containers[0].Env, "RESTRICT_RESOURCE_DELETION").Value = "true"
```

Comment:
to modify the `RESTRICT_RESOURCE_DELETION` config you need to add the `CONFIG_JS_` prefix in the env.
 ```suggestion
		util.GetEnvVariable(&r.CoreApp.Spec.Template.Spec.Containers[0].Env, "CONFIG_JS_RESTRICT_RESOURCE_DELETION").Value = "true"
```

Reply:
ack, changed.